Description of distributed_product_mode was added.

BayoNet opened the pull request.

Commit:
Information on <allow_databases> was added.

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Description of distributed_product_mode was added.
Many syntax errors was fixed which led to the warnings and errors of sphinx-build.

Commit:
Cross-link for distributed_product_mode wad added

robot-metrika-test commented:
Can one of the admins verify this patch?

Tri0L commented on docs/ru/query_language/queries.rst:
```diff
@@ -627,7 +649,7 @@ SELECT
 ``ARRAY JOIN`` - это, по сути, ``INNER JOIN`` с массивом. Пример:
 
-.. code-block:: sql
+::
```
Не совсем верно убирать code-block, в некоторых рендерах мы теряем подсветку SQL синтаксиса.

BayoNet replied:
А я как раз таки потому и убрал, что у меня подсветки не было :-).
Строка  ":) CREATE TABLE arrays_test (s String, arr Array(UInt8)) ENGINE = Memory" не распознается Сфинксовским хайлайтером как sql даже несмотря на прямое указание code-block:: sql. Видимо это происходит из-за символов ":)" в начале строки. Я предпочел убрать code-block, чем менять оригинальный текст.

alexey-milovidov commented on docs/ru/settings/settings.rst:
```diff
@@ -1,3 +1,29 @@
+------------------------
+Изменяет поведение :ref:`распределенных подзапросов <queries-distributed-subrequests>`.
+
+ClickHouse применяет настройку в том случае, когда в подзапросах на любом уровне встретилась распределенная таблица, которая существует на локальном сервере и имеет больше одного шарда.
```
Только подзапросы для IN, JOIN.

Только если в секции FROM используется распределённая таблица.
Именно поэтому настройка называется distributed_**product**_mode - настройка поведения в случае, если запрос содержит произведение распределённых таблиц.

Не используется в случае табличной функции remote.

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Description of <allow_databes> was improved.
Description of distributed_product_mode was improved.
Some decorative edits was made.

Commit:
Descriptions of `source/source_type/where` and `structure/attribute/expression` are added.

Commit:
Lagging replicas processing is described.
Desctiption of settings application is actualized.

Commit:
1. GraphiteMergeTree is added.
2. Merge remote-tracking branch 'upstream/master'.
